### PR TITLE
Fix 648: test for this.proxy before trying toString on it

### LINKED ISSUE #648: TypeError in sinon.js: 'toString' of undefined
Hey,

sinon version: 1.12.2
karma sinon: 1.0.4

Sinon works fine, but when i console log the window (console.log(window)) it throws the following error:

````
  Uncaught TypeError: Cannot read property 'toString' of undefined
  at /Users/kirrg/dev/avuba/avb-app-v2/node_modules/sinon/pkg/sinon.js:2138
````

`````
  toString: function () {
     var callStr = this.proxy.toString() + "(";
`````

this.proxy is undefined. how can i fix this?
i couldnt find a duplicate, if it is, just delete 'me'.

thank you!
kate

### PR DESCRIPTION
**Pull request by mroderick**
Fixes #648

I am not sure how we can test this, as it's not an error that shows up in our test suite.

The PR merely implements the hotfix that people have discovered and reported in #648

### CODE REVIEW

File: lib/sinon/call.js [hunk 1]
```diff
@@ -155,7 +155,7 @@
             },
 
             toString: function () {
-                var callStr = this.proxy.toString() + "(";
+                var callStr = this.proxy ? this.proxy.toString() + "(" : "";
                 var args = [];
 
                 for (var i = 0, l = this.args.length; i < l; ++i) {
```
Comment (fatso83) [inline lib/sinon/call.js]: You forgot [the other part](https://github.com/cjohansen/Sinon.JS/issues/648#issuecomment-118257844) of the hotfix:
```javascript
 if (!this.args){
   return ":(";
 }
```
Without it this breaks with 
```
	TypeError: 'undefined' is not an object (evaluating 'this.args.length')
	    at /Users/carl-erik.kopseng/dev-priv/Sinon.JS/pkg/sinon.js:2338
        ...
```
([ref](https://github.com/cjohansen/Sinon.JS/issues/648#issuecomment-139385678))

**Comment by joscha** [comment]
if I rebase this and fix the conflicts, is there a chance we can get this merged?

**Comment by fatso83** [comment]
Not sure how this went under the rader for so long. Will look into it in an hour, and I'll try to get it merged asap.

**Comment by fatso83** [comment]
I have rebased it and merged it to master (as 7a18eb5).

**Comment by fatso83** [comment]
@mroderick I am not sure, but I don't think this has been included in the 1.17-branch. 

Maybe I should make a separate PR for that? I see that some people would like a release of this, but 2.0 is not ready yet.

**Comment by joscha** [comment]
If we can port this to 1.17 that would be great.

**Comment by joscha** [comment]
backport is here: #970

**Pull request closed without merging**